pkg/action/pull.go

```diff
@@ -109,13 +109,13 @@ func (p *Pull) Run(chartRef string) (string, error) {
 		if !filepath.IsAbs(ud) {
 			ud = filepath.Join(p.DestDir, ud)
 		}
-		if fi, err := os.Stat(ud); err != nil {
+		if _, err := os.Stat(ud); err != nil {
```

Comment:
Probably need to refactor the code here if we want to check if that directory exists. This code originally just checked if the `destination/untardir` existed and created if it did not. For this PR we need to also account for what subdirectory will be extracted from the untar. 

For example, running command below:

```console
$ pwd
/home/usr1/test/helm-charts

$ ls -lrt

drwxr-xr-x  4 root root  4096 Nov 27 13:21 foo
[....]
drwxr-xr-x  4 root root  4096 Dec 16 11:17 chrt-7210

$ helm fetch stable/mysql --untar

$ ls -lrt

[...]
drwxr-xr-x  4 root root  4096 Dec 16 11:17 chrt-7210
drwxr-xr-x  3 root root  4096 Dec 16 12:25 mysql
```

I specify no directory for flags `--destination` or `--untardir`, it will use the current directory which is `.` in Linux and append the `<package_directory>`  which is `mysql`. Therefore need to check for the `<directory_path>/<package_directory>` because it is the `<package_directory>` which will cause the conflict within the directory path.

Reply:
Got it. I will fix it soon.